### PR TITLE
[DT-385] OBB files installation support

### PR DESCRIPTION
**Pull request by G0retZ**
**What does this PR do?**

   Add support for OBB files installation. 

   Some refactoring and improvement were performed in order to add this support.

**Database changed?**

   No

**Where should the reviewer start?**

- [ ] By commits

**How should this be manually tested?**

  Flow on how to test this or QA Tickets related to this use-case: [APP-385](https://aptoide.atlassian.net/browse/APP-385)
  Essentially try to install a game with OBBs and play around with the permissions

**What are the relevant tickets?**

  Tickets related to this pull-request: [APP-385](https://aptoide.atlassian.net/browse/APP-385)

**What are the relevant PRs?**

  PRs related to this one: [#239](https://github.com/Aptoide/aptoide-client-dt/pull/239)

**Code Review Checklist**

- [ ] Documentation on public interfaces
- [ ] Database changed? If yes - Migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] New Kotlin code has unit tests
- [ ] New flows in presenters unit tests
- [ ] Mappers/Validators with any kind of logic unit tests
- [ ] Functional tests pass

[APP-385]: https://aptoide.atlassian.net/browse/APP-385?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[APP-385]: https://aptoide.atlassian.net/browse/APP-385?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ricardo-frango** [comment]
it was failing on the first installation when install permission was not granted. if the app is restart the install permission granted, it works fine

**Review by ricardo-frango — APPROVED** [review]

File: download-view/src/main/java/cm/aptoide/pt/download_view/presentation/DownloadView.kt [hunk 1]
```diff
@@ -309,8 +290,12 @@ fun DownloadState(
     )
 
     DownloadUiState.Installed -> OpenButton(onOpenClick)
-    DownloadUiState.Error -> ErrorDownloadView()
-    DownloadUiState.ReadyToInstall -> ReadyToInstallView()
+    DownloadUiState.Error -> ErrorDownloadView(onInstallClick)
+    DownloadUiState.ReadyToInstall -> IndeterminateDownloadView(
+      label = "Waiting for install",
```
[the remark this answers is not in the document]
Comment (G0retZ) [reply download-view/src/main/java/cm/aptoide/pt/download_view/presentation/DownloadView.kt:295]: It is for future change

File: aptoide-installer/src/main/java/cm/aptoide/pt/installer/presentation/UserActionDialog.kt [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+    }
+  }
+}
+
+private val UserConfirmation.rationale
+  get() = when (this) {
+    UserConfirmation.INSTALL_SOURCE -> "By default system doesn't allow Aptoide to install apps. You need to mark Aptoide as the installation source first."
+    UserConfirmation.WRITE_EXTERNAL_RATIONALE,
+    UserConfirmation.WRITE_EXTERNAL,
+    -> "In order to install the required game resources allow Aptoide to access external storage."
```
[the remark this answers is not in the document]
Comment (G0retZ) [reply aptoide-installer/src/main/java/cm/aptoide/pt/installer/presentation/UserActionDialog.kt:113]: It is for future change

File: aptoide-installer/src/main/java/cm/aptoide/pt/installer/presentation/UserActionDialog.kt [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+          onClick = { viewModel.onResult(true) },
+        )
+      }
+    }
+  }
+}
+
+private val UserConfirmation.rationale
+  get() = when (this) {
+    UserConfirmation.INSTALL_SOURCE -> "By default system doesn't allow Aptoide to install apps. You need to mark Aptoide as the installation source first."
```
[the remark this answers is not in the document]
Comment (G0retZ) [reply aptoide-installer/src/main/java/cm/aptoide/pt/installer/presentation/UserActionDialog.kt:110]: It is for future change

File: aptoide-installer/src/main/java/cm/aptoide/pt/installer/presentation/UserActionDialog.kt [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+          title = "Ok",
+          onClick = { viewModel.onResult(false) },
+        )
+      } else {
+        DialogButton(
+          title = "Cancel",
+          onClick = { viewModel.onResult(false) },
+        )
+        DialogButton(
+          title = "Settings",
```
[the remark this answers is not in the document]
Comment (G0retZ) [reply aptoide-installer/src/main/java/cm/aptoide/pt/installer/presentation/UserActionDialog.kt:100]: It is for future change

File: aptoide-installer/src/main/java/cm/aptoide/pt/installer/presentation/UserActionDialog.kt [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+  (state as? ConfirmationAction)?.let {
+    PermissionsContent(rationale = it.confirmation.rationale) {
+      if (it.confirmation == UserConfirmation.WRITE_EXTERNAL_RATIONALE) {
+        DialogButton(
+          title = "Ok",
+          onClick = { viewModel.onResult(false) },
+        )
+      } else {
+        DialogButton(
+          title = "Cancel",
```
[the remark this answers is not in the document]
Comment (G0retZ) [reply aptoide-installer/src/main/java/cm/aptoide/pt/installer/presentation/UserActionDialog.kt:96]: It is for future change

File: aptoide-installer/src/main/java/cm/aptoide/pt/installer/presentation/UserActionDialog.kt [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+        else -> Unit
+      }
+    }
+  )
+
+  (state as? ConfirmationAction)?.let {
+    PermissionsContent(rationale = it.confirmation.rationale) {
+      if (it.confirmation == UserConfirmation.WRITE_EXTERNAL_RATIONALE) {
+        DialogButton(
+          title = "Ok",
```
[the remark this answers is not in the document]
Comment (G0retZ) [reply aptoide-installer/src/main/java/cm/aptoide/pt/installer/presentation/UserActionDialog.kt:91]: It is for future change

**Pull request merged**